doc/docusaurus/docs/auction-smart-contract/end-to-end/generating-keys.md

```diff
@@ -0,0 +1,44 @@
+---
+sidebar_position: 5
+---
+
+# Generating Keys and Addresses
+
+To start, clone the plutus-tx-template repository into the `on-chain` directory.
+Make sure to have [NodeJS](https://nodejs.org/en) and [yarn](https://yarnpkg.com/) (or [npm](https://github.com/npm/cli), which comes bundled with NodeJS) installed. Then, create a separate `off-chain` directory, set up `package.json`, and add the required dependencies:
+
+```
+git clone git@github.com:IntersectMBO/plutus-tx-template.git on-chain
+mkdir off-chain && cd $_
+yarn init -y
+yarn add @meshsdk/core
+yarn add cbor
+```
+
+We recommend using the Nix shell that comes with `plutus-tx-template` to run this example.
+The Nix shell provides the correct versions of all dependencies, including GHC, Cabal, Node.js, and various C libraries.
+To enter the nix shell, run
+
+```
+nix develop on-chain/
+```
+
+The first run of `nix develop` may take some time so please be patient.
+
+We'll use [mesh](https://meshjs.dev/), a JavaScript framework, for writing off-chain code.
+We'll use [Blockfrost](https://blockfrost.io/) as the blockchain provider, to avoid the need of running a local node.
```

Comment:
1. I have created a blockfrost project for the pre-prod network.
2. Its not clear if the Faucet mentioned on the `getting-funds.md` page uses preprod or preview net.
3. When trying to send funds I've got an error because blockfrost was running preprod but funds were on preview.

I suggest to add following info:
1. Instruct users to create a blockfrost project for the Preview network specifically.
2. Optionally: suggest checking funding transaction created by the faucet in a blockchain explorer, e.g. https://preview.cardanoscan.io/transaction/fab8d19b1c69c8b4556b7fb77da0d3648c09e129beeaae8b5ca8135dde54d645